lms/models/module_item_configuration.py

```diff
@@ -50,3 +50,11 @@ class ModuleItemConfiguration(BASE):
         server_default=sa.text("'{}'::jsonb"),
         nullable=False,
     )
+
+
+class CanvasModuleItemConfiguration(ModuleItemConfiguration):
+    def get_mapped_file_id(self, file_id):
+        return self.extra.get("canvas_file_mappings", {}).get(file_id)
+
+    def set_mapped_file_id(self, file_id, mapped_file_id):
+        self.extra.setdefault("canvas_file_mappings", {})[file_id] = mapped_file_id
```

Comment:
I've put this in a `CanvasModuleItemConfiguration` subclass in order to head off the objection that we can't have Canvas-specific methods on the `ModuleItemConfiguration` class. It means that if you want to use these two methods you have to do `db.query(CanvasModuleItemConfiguration)` not `db.query(ModuleItemConfiguration)`.

I considered just storing the mappings in a separate table `models.CanvasFileMapping`) instead of using a JSON column. But the service method that actually needs to use these mappings alreadys needs to have the `ModuleItemConfiguration` object for other reasons anyway. If the mappings are stored in a different table then the service has to do an extra query to get them, instead of just doing `mic.get_mapped_file_id()` and `mic.set_mapped_file_id()` when it already has `mic`.

These problems can be alleviated. If using a separate table you can use SQLAlchemy relations so that `mic.canvas_file_mappings` is a list of all the `CanvasFileMappings`. And I suspect you might also be able to use eager loading to avoid the additional SQL query.

But it's actually simpler to use JSON, and requires less fancy SQLAlchemy stuff (relations, eager loading).

Reply:
Not sure if that's the right place for those two method then, having both `db.query(CanvasModuleItemConfiguration)` and `db.query(ModuleItemConfiguration)` will suggest there's some difference at the storage layer.

I think we should either rename them to "get_canvas_mapped_file_id" and leave them on `ModuleItemConfiguration` (the model is storing canvas specific information, nothing wrong with retrieving that information in the same place IMO) or maybe move them to the service? 


Agree on JSON vs an extra table here.

Reply:
Ok, I've changed this to just `ModuleItemConfiguratin.{get,set}_canvas_file_mapping()`